pkg/opni/commands/debug.go

```diff
@@ -462,6 +467,85 @@ func BuildDebugImportAgentCmd() *cobra.Command {
 	return cmd
 }
 
+func BuildDebugAgentLogStreamGetCmd() *cobra.Command {
+	var since, until, level, output string
+	var follow bool
+	var names []string
+	cmd := &cobra.Command{
+		Use:   "agent-logs <cluster-id>",
+		Short: "Get agent logs",
+		Args:  cobra.ExactArgs(1),
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return completeClusters(cmd, args, toComplete)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				cl, err := mgmtClient.ListClusters(cmd.Context(), &managementv1.ListClustersRequest{})
+				if err != nil {
+					lg.Error("fatal", err)
+					os.Exit(1)
+				}
+				for _, c := range cl.Items {
+					args = append(args, c.Id)
+				}
+			}
+			streamRequest := &managementv1.StreamAgentLogsRequest{
+				Agent: &corev1.Reference{
+					Id: args[0],
+				},
+				Request: &controlv1.LogStreamRequest{
+					Filters: &controlv1.LogStreamFilters{
+						NamePattern: names,
+						Level:       lo.ToPtr(int32(logger.ParseLevel(level))),
+					},
+					Follow: follow,
+				},
+			}
+
+			startTime := parseTimeOrDie(since)
+			endTime := parseTimeOrDie(until)
+			streamRequest.Request.Since = timestamppb.New(startTime)
+			streamRequest.Request.Until = timestamppb.New(endTime)
+
+			stream, err := mgmtClient.GetAgentLogStream(cmd.Context(), streamRequest)
+			if err != nil {
+				return err
+			}
+			for {
+				log, err := stream.Recv()
+				if err != nil {
+					return err
+				}
+
+				done := (proto.Equal(log, &controlv1.StructuredLogRecord{}))
+				keepFollowing := done && follow
+				if keepFollowing {
+					continue
+				} else if done {
+					return nil
+				}
+
+				var attrs strings.Builder
+				for _, attr := range log.Attributes {
+					attrs.WriteString(attr.Key)
+					attrs.WriteString("=")
+					attrs.WriteString(attr.Value)
+					attrs.WriteString(" ")
+				}
+				fmt.Println(log.Time.AsTime().Format(logger.DefaultTimeFormat), log.Level, log.Name, log.Source, log.Message, attrs.String())
```

Comment:
Should this use `slog.Log(...)`?

Reply:
Thanks, this reminded me I need to implement `output=json` - I would use `slog.Log` there, eg
```
handlerOptions := &slog.HandlerOptions{
		Level: logger.ParseLevel(level), 
		AddSource: false, 
		ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
		if a.Key == slog.TimeKey && len(groups) == 0 {
			return slog.Attr{}
		}
		return a
		},
	}

if output == "json" {
      slog.SetDefault(slog.New(slog.NewJSONHandler(os.Stderr, handlerOptions)))
}
...

slog.Log(context.TODO(), logger.ParseLevel(log.GetLevel()), log.GetMessage(), attrs...)

```

For `output=text` though, if I use `slog.SetDefault(logger.New())`, I'm unable to overwrite the `<time user called opni debug> ` timestamp while preserving the log formatting. I could write a handler that assumes there will be a `time` attribute in every call to slog.Log and replace the original timestamp, except I don't want to rely on that assumption.

Here are some alternatives to `fmt.Println`. What do you think?

Option 1: preserve tinted colors but leave the double timestamp
```
slog.SetDefault(logger.New(logger.WithDisableCaller()))
```
![image](https://github.com/rancher/opni/assets/84587295/6b4a6abd-926f-4387-a135-46ad664f29e4)

Option 2: use default slog TextHandler, and remove the debug cli time/source
```
slog.SetDefault(slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: logger.ParseLevel(level), AddSource: false, ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
					if a.Key == slog.TimeKey && len(groups) == 0 {
						return slog.Attr{}
					}
					return a
				},
				})))
```
<img width="1531" alt="image" src="https://github.com/rancher/opni/assets/84587295/bb315748-c3c4-449b-8fb0-3f4a345a38c6">